woocommerce/woocommerce-ios
# Google Ads: Show campaign list from hub menu if there are existing campaigns

Closes: #14631 


## Description

When smoke testing the 21.2 build, I found that the entry point in the hub menu always navigates to the campaign creation page even though my store has existing campaigns. This PR addresses the issue by fixing the incorrect logic check for `shouldStartCampaignCreation`.

Note: If the entry point is tapped before the fetching is done, the campaign list is still empty and will navigate to campaign creation. I thought about hiding the entry point while fetching but this makes the UX worse, so I reverted the change.

Also: fix another wrong logic about refreshing Blaze eligibility in Hub Menu.

## Steps to reproduce


**Testing Google Ads**
- Pre-requisite: Install, activate, and set up the Google Ads plugin to a test site following pe5sF9-2Qo-p2. Ensure that you're using the plugin version 2.7.5 or above, and that the store has no campaign.
- Build the app and log in to your store.
- Navigate to the Menu tab.
- Tap on the entry point and confirm that the campaign creation page is displayed.
- Create a new campaign for the store.
- Quit the app and open again.
- Navigate to the Menu tab and wait for a few seconds before tapping Google Ads (to ensure that fetching campaigns are completed). Confirm that the campaign list is displayed.

**Testing Blaze**
- Log in to a self-hosted store with Jetpack connected.
- Navigate to Menu tab and confirm that the Blaze entry point is not available.
- Install Blaze Ads plugin on the site.
- Navigate away then back to the Menu tab. Confirm that the Blaze entry point is then available.

## Testing information

Tested on simulator iPhone 16 Pro iOS 18.1 and confirmed that the entry point to Google Ads now works correctly. Blaze entry point is updated after the site is eligible.

## Screenshots


https://github.com/user-attachments/assets/2a5c286d-f27a-43be-afbf-48b3dd07b7fb



---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
